### ISSUE TITLE
[css-pseudo-4] Add CSSPseudoElement.parentElement

### ISSUE THREAD

**Issue by birtles** [w3c/csswg-drafts#2816]
A `CSSPseudoElement` would be much more useful if it were possible to cross-reference it to the DOM tree. I believe this was part of the `PseudoElement` specified in CSSOM before that disappeared.

(In Gecko we currently implement `CSSPseudoElement` with only two attributes: `type` and `parentElement` since that seems to be all we need for now.)

**Comment by Loirooriol** [comment]
Is this "parent element" the [originating element](https://drafts.csswg.org/selectors-4/#originating-element), or is this a different concept?

**Comment by AmeliaBR** [comment]
I agree with @Loirooriol that "parent element" is a potentially confusing name for "originating element", because of newer pseudo-elements like ::part (or the various well-established, prefixed selectors for styling browser widgets) which might match an element inside a shadow tree, where that element has parent elements in the shadow structure.

**Comment by birtles** [comment]
Yes, I think originating element is the corresponding spec term. I'm not sure what the IDL member should be called.

**Comment by fantasai** [comment]
Maybe just .element?

**Comment by tabatkins** [comment]
Hmm yeah, `.parentElement` is compelling but wrong; `.originatingElement` is correct but soft-exceeds my complexity budget for spelling.  `.element` sounds good.

**Comment by emilio** [comment]
Gecko is not shipping this on any release channel other than Nightly: https://searchfox.org/mozilla-central/rev/bee8cf15c901b9f4b0c074c9977da4bbebc506e3/modules/libpref/init/StaticPrefList.h#137
